purchase_order_move_request/models/purchase_order.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    move_request_ids = fields.One2many(
+        comodel_name='account.move.request',
+        inverse_name='purchase_id',
+        string='Move Requests',
+    )
+    move_request_count = fields.Integer(
+        string='Move Request Count',
+        compute='_compute_move_request_count',
+    )
+
+    @api.depends('move_request_ids')
+    def _compute_move_request_count(self):
+        for order in self:
+            order.move_request_count = len(order.move_request_ids)
+
+    def _prepare_move_request_line_vals(self, line):
+        account = (
+            line.product_id.property_account_expense_id
+            or line.product_id.categ_id.property_account_expense_categ_id
+        )
+        return {
+            "product_id": line.product_id.id,
+            "name": line.name,
+            "quantity": line.product_qty,
+            "price_unit": line.price_unit,
+            "account_id": account.id if account else False,
+            "tax_ids": [Command.set(line.taxes_id.ids)],
+            "analytic_distribution": line.analytic_distribution,
+        }
+
+    def _prepare_move_request_vals(self):
+        return {
+            "purchase_id": self.id,
+            "partner_id": self.partner_id.id,
+            "line_ids": [
+                Command.create(self._prepare_move_request_line_vals(line))
```

Comment:
ย้าย method ข้างบนก่อน

แล้วเรียกใช้ method แบบนี้
```suggestion
                Command.create(line._prepare_move_request_line_vals())
```